app/scripts/controllers/account-order.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  BaseControllerV2,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+
+// Unique name for the controller
+const controllerName = 'AccountOrderController';
+
+/**
+ * The account ID of a account.
+ */
+export type AccountAddress = string;
+
+// State shape for AccountOrderController
+export type AccountOrderControllerState = {
+  pinnedAccountList: AccountAddress[];
```

Comment:
As @k-g-j mentioned, I noticed that the pinned account order is based on the account name. Should we add a timestamp here so that we can sort then by when they were pinned? Then we could pin them based on the latest pinned account.